src/commands/distribute/lib/mc-fus-uploader/mc-fus-uploader.ts

```diff
@@ -0,0 +1,764 @@
+
+    // After all checks have completed finally proceed
+    // to initialize all the upload required fields.
+    setState(McFusUploadState.New);
+
+    // Initialize all retry flags for the new upload.
+    uploadStatus.AutoRetryCount = 3;
+    uploadStatus.BlocksCompleted = 0;
+    uploadStatus.ServiceCallback.AutoRetryCount = 5;
+    uploadStatus.ServiceCallback.AutoRetryDelay = 1;
+    uploadStatus.ServiceCallback.FailureCount = 0;
+
+    // Copy all the required settings on to the upload data.
+    uploadData.AssetId = settings.AssetId;
+    uploadData.UploadDomain = settings.UploadDomain;
+    uploadData.UrlEncodedToken = settings.UrlEncodedToken;
+    uploadData.Tenant = settings.Tenant;
+
+    log("Upload created");
+  }
+
+  function invokeCallback(location: string) {
+    if (uploadData.CallbackUrl && uploadData.CallbackUrl !== "") {
+      const callbackUrl =
+        uploadData.CallbackUrl +
+        "/" +
+        encodeURIComponent(uploadData.AssetId) +
+        "?file_name=" +
+        encodeURIComponent(uploadData.File!.fileName) +
+        "&file_size=" +
+        encodeURIComponent(uploadData.File!.size) +
+        "&location=" +
+        location;
+      log("Callback was supplied. Invoking callback on: " + callbackUrl);
+
+      sendRequest({
+        type: "POST",
+        url: callbackUrl,
+        useBaseDomain: false,
+        useAuthentication: false,
+        error: function (err: Error) {
+          const errorMessage = "Callback failed. Status: " + err.message;
+
+          // Non-fatal error, just log info
+          log(errorMessage, { FailedCallback: uploadData.CallbackUrl });
+
+          // If we still have retries available go ahead with the success callback.
+          if (uploadStatus.ServiceCallback.AutoRetryCount > 0) {
+            setTimeout(function () {
+              invokeCallback(location);
+            }, uploadStatus.ServiceCallback.AutoRetryDelay * 10);
+
+            uploadStatus.ServiceCallback.AutoRetryCount--;
+            uploadStatus.ServiceCallback.FailureCount++;
+
+            // Increment the backoff in multiples of 5 for
+            // subsequent attempts. (5, 25, 125 and so on)
+            uploadStatus.ServiceCallback.AutoRetryDelay *= 5;
+          } else {
+            log(
+              "Callback retries depleted. The upload completed but the uploader was unable to perform a successful callback notifying completion."
+            );
+          }
+        },
+        success: function () {
+          log("Callback succeeded.");
+        },
+      });
+    }
+  }
+
+  function isUploadInProgress() {
+    return (
+      uploadStatus.State === McFusUploadState.Initialized ||
+      uploadStatus.State === McFusUploadState.Uploading ||
+      uploadStatus.State === McFusUploadState.Verifying
+    );
+  }
+
+  function isValidChunk(chunk: Buffer): boolean {
+    if (!chunk) {
+      return false;
+    }
+    if (chunk.length === 0) {
+      return false;
+    }
+    return true;
```

Comment:
I guess something like this should work here
```suggestion
    return chunk && chunk.length > 0;
```